alacritty/src/display.rs

```diff
@@ -207,8 +209,19 @@ impl Display {
         });
 
         // We should call `clear` when window is offscreen, so when `window.show()` happens it
-        // would be with background color instead of uninitialized surface.
-        window.swap_buffers();
+        // would be with background color instead of uninitialized surface. On Wayland we can
+        // safely ignore this call, since the window isn't visible until you actually draw
+        // something into it.
+        #[cfg(not(any(target_os = "macos", windows)))]
+        {
+            if event_loop.is_x11() {
+                window.swap_buffers()
+            }
+        }
+        #[cfg(any(target_os = "macos", windows))]
+        {
+            window.swap_buffers()
+        }
```

Comment:
Honestly I don't think we need this on macOS/Windows at all, right? I can test it on macOS to make sure, but originally this was just introduced for X11.

Reply:
Ehm, they also have a concept of window before drawing, so you should just rely, that your compositor doing something that make sense. I feel like we should just leave it as it is.

Reply:
Not all platforms require that you manually clear the window yourself though. Some platforms might make it invisible, which works fine. On macOS, it works great for example and I don't think there has ever been an issue with that.

So why should we potentially delay the startup significantly (remember swap_buffers might block), when it has no benefit?